src/poetry/console/commands/check.py

```diff
@@ -19,6 +25,32 @@ def handle(self) -> int:
         poetry_file = Factory.locate(Path.cwd())
         config = PyProjectTOML(poetry_file).poetry_config
         check_result = Factory.validate(config, strict=True)
+
+        unrecognized = sorted(
+            set(config.get("classifiers", []))
+            - set(classifiers)
+            - set(deprecated_classifiers)
+        )
+        if unrecognized:
+            check_result["errors"].append(
+                f"Unrecognized classifiers: {unrecognized!r}."
+            )
+
+        deprecated = sorted(
+            set(config.get("classifiers", [])).intersection(set(deprecated_classifiers))
+        )
+        if deprecated:
+            for old_classifier in deprecated:
+                message = f"Deprecated classifier {old_classifier!r}."
+                new_classifiers = cast(
+                    Dict[str, List[str]], deprecated_classifiers[old_classifier]
+                )
+                if new_classifiers:
+                    message += f" Must be replaced by {new_classifiers!r}."
```

Comment:
I'd rather build a complete sentence than the multiple here.

Reply:
Fixed in: https://github.com/python-poetry/poetry/pull/2881/commits/d940d2440a323afe77c275037d1c7fdd53f6a0f5